### ISSUE TITLE
hi just reporting another bug 

### ISSUE THREAD

**Issue by grant2258** [tknorris/tknorris-beta-repo#36]
Sorry thats all i seem to do on here is report bugs. I just want to leave some details of the bug. I got this error. I just done a grep "thread-31". because the tail -f showed thread 31 committed suicide and python script stopped.  

17:44:28 8215.848633 T:1641018272   DEBUG: Stream All The Sources: Pool Manager: Thread-31 thrown in Pool: (30/None)
17:44:28 8215.855469 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 handling job: |<function parallel_get_sources at 0x5e57c6b0>| with args: |[<scrapers.premiumize_scraper.Scraper object at 0x5e9118d0>, <scrapers.ScraperVideo instance at 0x5e37a990>]| and kwargs: |{}|
17:44:29 8216.585938 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 handling job: |<function parallel_get_sources at 0x5e57c6b0>| with args: |[<scrapers.crazy_scraper.Scraper object at 0x5e911250>, <scrapers.ScraperVideo instance at 0x5e37a990>]| and kwargs: |{}|
17:44:30 8218.372070 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 handling job: |<function parallel_get_sources at 0x5e57c6b0>| with args: |[<scrapers.vivoto_scraper.Scraper object at 0x5e909250>, <scrapers.ScraperVideo instance at 0x5e37a990>]| and kwargs: |{}|
17:44:46 8233.831055 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31
17:44:46 8233.831055 T:1624241056   DEBUG: Stream All The Sources: Worker Thread-31 still running
17:45:13 8261.507812 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31
17:45:13 8261.507812 T:1624241056   DEBUG: Stream All The Sources: Worker Thread-31 still running
17:45:19 8267.302734 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 committing suicide.
17:57:05 8972.956055 T:1310876576   DEBUG: Stream All The Sources: Pool Manager: Thread-31 thrown in Pool: (30/None)
17:57:13 8980.711914 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31
17:57:13 8980.712891 T:1624241056   DEBUG: Stream All The Sources: Worker Thread-31 still running
17:57:13 8980.739258 T:621573024   DEBUG: Stream All The Sources: Worker: Thread-31 committing suicide.
17:57:30 8998.067383 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31

The symptoms are sometimes kodi crashes other times it goes back as finding no sources other times in crashes and looses all kodi settings. I disabled the three scrappers and all seems good now just in case this helps anyone having problems.

**Comment by grant2258** [comment]
another thing i forgot to mention Im using the latest openelec 7.0 beta compiled myself and am using kodi 16.1. I have the same symptoms on openelec 6.0.3 as well.

**Comment by tknorris** [comment]
There is nothing in that log that is abnormal.

**Comment by tknorris** [comment]
Which scraper did you disable?

**Comment by grant2258** [comment]
sorry that was a bad paste i missed the top lines

less  kodi.log | grep "Thread-31"
17:44:28 8215.848633 T:1641018272   DEBUG: Stream All The Sources: Pool Manager: Thread-31 thrown in Pool: (30/None)
17:44:28 8215.855469 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 handling job: |<function parallel_get_sources at 0x5e57c6b0>| with args: |[<scrapers.premiumize_scraper.Scraper object at 0x5e9118d0>, <scrapers.ScraperVideo instance at 0x5e37a990>]| and kwargs: |{}|
17:44:29 8216.585938 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 handling job: |<function parallel_get_sources at 0x5e57c6b0>| with args: |[<scrapers.crazy_scraper.Scraper object at 0x5e911250>, <scrapers.ScraperVideo instance at 0x5e37a990>]| and kwargs: |{}|
17:44:30 8218.372070 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 handling job: |<function parallel_get_sources at 0x5e57c6b0>| with args: |[<scrapers.vivoto_scraper.Scraper object at 0x5e909250>, <scrapers.ScraperVideo instance at 0x5e37a990>]| and kwargs: |{}|
17:44:46 8233.831055 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31
17:44:46 8233.831055 T:1624241056   DEBUG: Stream All The Sources: Worker Thread-31 still running
17:45:13 8261.507812 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31
17:45:13 8261.507812 T:1624241056   DEBUG: Stream All The Sources: Worker Thread-31 still running
17:45:19 8267.302734 T:1310876576   DEBUG: Stream All The Sources: Worker: Thread-31 committing suicide.
17:57:05 8972.956055 T:1310876576   DEBUG: Stream All The Sources: Pool Manager: Thread-31 thrown in Pool: (30/None)
17:57:13 8980.711914 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31
17:57:13 8980.712891 T:1624241056   DEBUG: Stream All The Sources: Worker Thread-31 still running
17:57:13 8980.739258 T:621573024   DEBUG: Stream All The Sources: Worker: Thread-31 committing suicide.
17:57:30 8998.067383 T:1624241056   DEBUG: Stream All The Sources: Reaping: Thread-31

**Comment by grant2258** [comment]
its not pasting properly ill attach a file sorry about this
[kodi.txt](https://github.com/tknorris/tknorris-beta-repo/files/435401/kodi.txt)

**Comment by tknorris** [comment]
The paste is fine, but there's nothing abnormal in there. Again, what scrapers did you disable?

**Comment by grant2258** [comment]
scrapers.premiumize_scraper i disabled both versions
scrapers.crazy_scraper
scrapers.vivoto_scraper

ill try enable one at a time see if i can narrow it down

**Comment by tknorris** [comment]
it's not very likely that any of those could cause a problem...

**Comment by grant2258** [comment]
This is strange I have 3 sd cards openelec 6 and 7beta3 and the other one has the latest 7 compiled myself. Whats happening in all the versions is the same no response for a long time time kodi either crashes and goes back to episode listing or resets kodi and you loose the settings(it doesnt notify of time out). I tried the same thing on all 3 i disabled them 3 scrappers it works fine after doing that.

When I enable them again it works fine as well. This one has me scratching my head is it works when I disable and enable again.

This on a raspberry pi 2. One thing to note is i always get a timeout message from vivo.to after doing this intead of a crash or a dropback to episodes listing without any timeout notification.


to get it working.

I disable vivo.to crazy and premiumize  scrapers (thats cause the was the three that was listed) I clear cache(dont know if this is needed then i search a tv show everything is fine. I then enable the the scrapers again do a search for a tv show and it works, it shows a timeout instead of just falling back the the episode list or crashing for vivo.to now. 

I did try just clearing the cach that didnt work
